spec/v1.0-draft/README.md

```diff
@@ -109,29 +107,27 @@ subject and unambiguously identifying the types of the [predicate]. It is a
 > Identifier for the schema of the Statement. Always
 > `https://in-toto.io/Statement/v1.0` for this version of the spec.
 
-`subject` _array of objects, required_
+`subject` _array of objects([Reference], required_
 
 > Set of software artifacts that the attestation applies to. Each element
 > represents a single software artifact.
 >
-> IMPORTANT: Subject artifacts are matched purely by digest, regardless of
-> content type. If this matters to you, please comment on
-> [GitHub Issue #28](https://github.com/in-toto/attestation/issues/28)
+> IMPORTANT: Subject artifacts SHOULD be matched by digest or by uri,
+> regardless of MIME type.
```

Comment:
I was attempting to match the semantics of the SLSA v1 ArtifactReference with this change, for complete compatibility; it's quite possible I missed something on the SLSA side. If a digest is set, the digest should be used for matching, regardless of whether a URI is set or not. I will clarify this in the documentation.

Reply:
Update: I opened issue https://github.com/slsa-framework/slsa/issues/614 to discuss this further, because I'm really wondering about the `digest` vs `uri` matching semantics that seem to be implied in the current SLSA Provenance v1 ArtifactReference.